src/test/java/picard/analysis/AlignmentSummaryMetricsCollectorTest.java

```diff
@@ -22,6 +27,9 @@
                 {"20M2S", true, 0},
                 {"20M2S", false, 2},
 
+                {"20M2S4S", false, 6},
```

Comment:
Are these legal cigar strings? Doesn't the S operator have to come at the end? I guess it doesn't matter for this test and I agree with 6 for this length (and 0 for the one below).

Reply:
From the spec:
```
• H can only be present as the first and/or last operation.
• S may only have H operations between them and the ends of the CIGAR string
```

so probably, no. but I wanted to make the code robust to this....